### PR TITLE
feat(fmt): add `prettyDuration`

### PR DESCRIPTION
**Pull request by retraigo**
A smaller version of [duration.js](https://github.com/retraigo/duration.js) that only performs formatting milliseconds into strings.

### CODE REVIEW

File: fmt/README.md [hunk 1]
```diff
@@ -17,47 +17,47 @@ below.
 
 This is very much a work-in-progress. I'm actively soliciting feedback.
 
-- What useful features are available in other languages apart from Golang and C?
+-   What useful features are available in other languages apart from Golang and C?
```
Comment (cjihrig) [inline fmt/README.md:20]: Please undo all of the style only changes like this. It make the diff bigger and PRs more difficult to review.

File: fmt/numbers.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// A module to format numbers.
+
+/** Add zeros to the beginning of a number till it reaches a certain digit count. */
```
Comment (cjihrig) [inline fmt/numbers.ts:4]:
```suggestion
/** Add zeros to the beginning of a number until it reaches a certain digit count. */
```

File: fmt/numbers.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+export function addZero(num: number, digits = 3): string {
```
Comment (cjihrig) [inline fmt/numbers.ts:5]: Would something simpler like this work:

```js
number.toString().padStart(digits, '0')
```

File: fmt/numbers_test.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Deno.test({
+  name: "random number",
+  fn() {
+    const num = Math.round(Math.random() * 9787923);
+    assertEquals(num.toString(), addZero(num, num.toString().length));
+  },
+});
+
+Deno.test({
+  name: "bond",
```
Comment (cjihrig) [inline fmt/numbers_test.ts:15]:
```suggestion
  name: "pads the start of a number",
```

File: fmt/duration.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+import { addZero } from "./numbers.ts";
+
+interface DurationObj {
+  d: number;
+  h: number;
+  m: number;
+  s: number;
+  ms: number;
+  µs: number;
```
Comment (cjihrig) [inline fmt/duration.ts:12]: Can we use `us` instead.

File: fmt/README.md (hunk 1 above)
Comment (retraigo) [reply fmt/README.md]: done

File: fmt/numbers.ts (hunk 2 above)
Comment (retraigo) [reply fmt/numbers.ts]: removed the entire file because it was pointless

File: fmt/numbers.ts (hunk 3 above)
Comment (retraigo) [reply fmt/numbers.ts]: Yeah forgot padStart existed.

File: fmt/README.md [hunk 6]
```diff
@@ -268,6 +268,26 @@ prettyBytes(1337, { locale: "de" });
 //=> '1,34 kB'
 ```
 
+# Pretty Duration
+
+Get formatted time duration from milliseconds.
+
+```ts
+import { prettyBytes } from "https://deno.land/std@$STD_VERSION/fmt/duration.ts";
```
Comment (kt3k) [inline fmt/README.md:276]:
```suggestion
import { prettyDuration } from "https://deno.land/std@$STD_VERSION/fmt/duration.ts";
```

Comment (retraigo) [reply fmt/README.md:276]: My bad

**Review by cjihrig — APPROVED** [review]

**Review by kt3k — APPROVED** [review]
LGTM

Note: There's a very similar ES feature proposal https://github.com/tc39/proposal-intl-duration-format but it's unclear when it's available in V8. So I think it makes sense to include this to std until that feature is implemented in V8

**Comment by retraigo** [comment]
Yeah I'm looking forward to Temporals becoming part of the spec.

**Comment by timreichen** [comment]
As a succession PR with some enhancements: https://github.com/denoland/deno_std/pull/2871

**Pull request merged**